packages/tools/addon/services/cardstack-tools.js

```diff
@@ -48,10 +50,10 @@ export default Service.extend({
       });
       return fields;
     });
-    return modelFields.reduce((flattened, fields) => {
+    this.set('modelFields', modelFields.reduce((flattened, fields) => {
       return flattened.concat(fields);
-    });
-  }),
+    }));
+  }).observes('_renderedFieldNames', 'activeContentItem.model').on('init'),
```

Comment:
is there no other way to implement this that doesn't use observers?

Reply:
my understanding is that E-C doesn't use Ember observers under the hood, it has its own mechanism for handling this. Documentation for E-C `observes` is sparse, however, so don't hold me to that 🙈

Reply:
> my understanding is that E-C doesn't use Ember observers under the hood, it has its own mechanism for handling this

http://ember-concurrency.com/api/task-property.js.html indicate that it's using `import { addObserver } from '@ember/object/observers'` under the hood. which is actually a little worrying to me as it does not seem to import `removeObserver` in that file... 🤔

Reply:
hmmm this could explain why there are failing `tools` tests 😆I will investigate further.